Microsoft.HealthVault.Fhir/Transformers/MedicationToFhir.cs

```diff
@@ -0,0 +1,178 @@
+        {
+            fhirMedication.Code = hvMedication.Name.ToFhir();
+
+            if (hvMedication.GenericName != null)
+            {
+                var ingredientItem = hvMedication.GenericName.ToFhir();
+                fhirMedication.Ingredient = new List<FhirMedication.IngredientComponent> {
+                    new FhirMedication.IngredientComponent(){
+                        Item= ingredientItem
+                }};
+            }
+
+            if (hvMedication.Strength != null)
+            {
+                if (!fhirMedication.Ingredient.Any())
+                {
+                    var ingredientItem = hvMedication.Name.ToFhir();
+                    fhirMedication.Ingredient = new List<FhirMedication.IngredientComponent> {
+                        new FhirMedication.IngredientComponent{
+                            Item = ingredientItem
+                    }};
+                }
+
+                if (hvMedication.Strength.Structured.Any())
+                {
+                    var amount = HealthVaultCodesToFhir.GetSimpleQuantity(hvMedication.Strength);
+
+                    fhirMedication.Ingredient.First().Amount = new Ratio() { Numerator = amount };
+                }
+            }
+
+            return fhirMedication;
+        }
+
+        internal static MedicationRequest ToFhirInternal(Prescription prescription, MedicationRequest medicationRequest)
+        {
+            medicationRequest.SetIntentAsInstanceOrder();
+            //hvMedication.PrescribedBy
+            medicationRequest.AuthoredOnElement = prescription.DatePrescribed?.ToFhir();
+            if (prescription.AmountPrescribed != null
+                || prescription.Refills.HasValue
+                || prescription.DaysSupply.HasValue
+                || prescription.PrescriptionExpiration != null)
+            {
+                medicationRequest.DispenseRequest = new MedicationRequest.DispenseRequestComponent
+                {
+                    Quantity = HealthVaultCodesToFhir.GetSimpleQuantity(prescription.AmountPrescribed),
+                    NumberOfRepeatsAllowed = prescription.Refills == 0 ? null : prescription.Refills,
+                    ExpectedSupplyDuration = new Duration()
+                    {
+                        Value = prescription.DaysSupply,
+                        Code = nameof(UnitsNet.Units.DurationUnit.Day)
+                    },
+                    ValidityPeriod = new Period
+                    {
+                        EndElement = prescription.PrescriptionExpiration?.ToFhir()
+                    }
+                };
+            }
+            if (prescription.Substitution != null)
+            {
+                medicationRequest.Substitution = new MedicationRequest.SubstitutionComponent
+                {
+                    Allowed = IsAllowed(prescription.Substitution)
+                };
+            }
```

Comment:
If allowed is not present or cannot be transformed we probably should omit the Substitution element on the medication request. I will recommend you do not throw an exception on IsAllowed, check if the IsAllowed method is returning something different than null and then create the substitution component if you can fill it.